src/allmydata/web/common.py

```diff
@@ -387,6 +387,22 @@ def renderHTTP(self, ctx):
             raise server.UnsupportedMethod(getattr(self, 'allowedMethods', ()))
         return m(ctx)
 
+    def render_OPTIONS(self, ctx):
+        """
+        Handle HTTP OPTIONS request by adding appropriate headers.
+
+        :param ctx: client transaction from which request is extracted
+        :returns: str (empty)
+        """
+        from allmydata import __version__
+        req = IRequest(ctx)
+        req.setHeader("server", "Tahoe-LAFS gateway v" + __version__)
+        methods = ', '.join([m[7:] for m in dir(self) if m.startswith('render_')])
```

Comment:
Do we do this other places? I kind of think maybe we *shouldn't* stick the exact version in there (e.g. just "Tahoe-LAFS gateway")

Reply:
Not that I am aware of. I removed it.